testutil/testutil.go

```diff
@@ -0,0 +1,98 @@
+package testutil
+
+import (
+	"net"
+	"sync"
+)
+
+// Test constants
+const (
+	CloseOnMsgIdx = 5
+	MaxMsgCount   = 10
+)
+
+// LocalTCPServer tcp server for testing specific network errors
+type LocalTCPServer struct {
+	lnmu     sync.RWMutex
+	MockConn *Conn
```

Comment:
The approach of mocking net connections is actually not terribly idiomatic. Because it's relatively easy in Go, most network tests in Go actually stand up a full network client/server instead of building mocks. This reduces a lot of code and better represents reality.